source/GameAction.cpp

```diff
@@ -367,6 +449,11 @@ void GameAction::Do(PlayerInfo &player, UI *ui) const
 				player.FailMission(mission);
 	}
 
+	for(const auto &hardpoint : hardpoints)
+		player.FlagshipPtr()->AddHardpoint(hardpoint);
+	for(const auto &attribute : attributes)
+		player.FlagshipPtr()->AddOutfit(&attribute, 1);
```

Comment:
I'm really not sure this is going to do what you want it to do.

Reply:
In what way?

Reply:
I thought about a new function add stats instead?